lib/plugins/aws/package/compile/events/alb/lib/validate.js

```diff
@@ -180,4 +187,376 @@ module.exports = {
     }
     return event.alb.multiValueHeaders;
   },
+
+  // --------- AuthenticateCognitoConfig
+  validateCognitoAuth(event, functionName) {
+    if (!_.isObject(event.alb.authenticateCognito)) {
```

Comment:
Let's not implement such sophisticated validation here. Here let's limit it just to avoid eventual errors like _Cannot access x on undefined_ (so if some object is required let's crash meaningfully if it's not provided)

There are two reasons for that:
1. We're about to unleash schema based config validation https://github.com/serverless/serverless/issues/6562 so shortly any validation like that should be added just by extending the schema.
While implementing validation manually like proposed here, increases bloat of a project and adds to maintenance cost.
2. Generated template is well validated on AWS side which should be considered as the only source of truth (although it's definitely great to pick obvious errors before we pass template for AWS validation)

Reply:
Ok. So basically just to validation to make sure the required fields are given. For the rest default values are passed in cloud formation?

Reply:
Technically just validation of required object, on which you expect some properties.

So:
1. Do not validate `options.foo` if it's `options.foo` that we pass it directly to CF template (even if foo is required).
2. Validate `options.foo`(simply if it's truthy) if it's `options.foo.bar` that's required and we need to pass it to CF template (but don't validate `options.foo.bar`)